### PR TITLE
Glonass support, ubloxd add ephemeris parsing

### PR DESCRIPTION
**Pull request by gast04**
add glonass parsing and publishing to ubloxd, not enabled yet
diff is so large cause gps ephemeris parsing got moved to a new function

### CODE REVIEW

File: release/files_common [hunk 1]
```diff
@@ -229,6 +229,8 @@ selfdrive/locationd/generated/ubx.cpp
 selfdrive/locationd/generated/ubx.h
 selfdrive/locationd/generated/gps.cpp
 selfdrive/locationd/generated/gps.h
+selfdrive/locationd/generated/glonass.cpp
+selfdrive/locationd/generated/glonass.h
```
Comment (adeebshihadeh) [inline release/files_common:233]: you can replace all the generated/ lines here with
```suggestion
selfdrive/locationd/generated/*.cpp
selfdrive/locationd/generated/*.h
```

**Comment by adeebshihadeh** [comment]
> diff is so large cause gps ephemeris parsing got moved to a new function

can you first split that out into its own preparation PR to reduce the diff here?

**Comment by HaraldSchafer** [comment]
LGTM

File: release/files_common (hunk 1 above)
Comment (gast04) [reply release/files_common:233]: added in https://github.com/commaai/openpilot/pull/27125
which needs to be merged first

**Comment by gast04** [comment]
split up, merge oder
* https://github.com/commaai/openpilot/pull/27122
* https://github.com/commaai/openpilot/pull/27125

then  I will rebase this one

**Review by adeebshihadeh — APPROVED** [review]
NIce job, it's much cleaner after splitting out the null-effect GPS ephemeris parsing changes!

File: selfdrive/locationd/ublox_msg.cc [hunk 2]
```diff
@@ -246,11 +264,143 @@ kj::Array<capnp::word> UbloxMsgParser::parse_gps_ephemeris(ubx_t::rxm_sfrbx_t *m
...
+    eph.setSvType(data->m());
+  }
+
+  // string number 5
+  {
+    kaitai::kstream stream(glonass_strings[msg->sv_id()][5]);
+    glonass_t gl_stream(&stream);
+    glonass_t::string_5_t* data = static_cast<glonass_t::string_5_t*>(gl_stream.data());
+
+    // TODO: remove parsing string5, its only needed for the year
```
[the remark this answers is not in the document]
Comment (gast04) [reply selfdrive/locationd/ublox_msg.cc:376]: not sure if ever tbh, in theorie we can get the year later in python using datetime, but than we need to account leap year and rollovers and this was just not worth it in my mind to get the ephemeris 2seconds faster (time of receiving one string)

**Pull request merged**